src/crypto/hash.h

```diff
@@ -18,7 +18,7 @@ namespace crypto
     Sha256Hash();
     Sha256Hash(std::initializer_list<CBuffer> il);
 
-    std::array<uint8_t, SIZE> h;
+    std::array<uint8_t, SIZE> h = {};
```

Comment:
Why do we need to initialize this array every time we create a hash? Can we not only pay the cost when we need to initialize it?

Reply:
The array needs to be zeroe'd out for the genesis transaction, which queries the signature table for the root of the latest signature, but in genesis the signature table is empty, causing the default constructor to be called.

Reply:
discussed offline - I think this is fine